planner/filter/normalize_test.go

```diff
@@ -291,7 +291,8 @@ func TestNormalizeConditions(t *testing.T) {
 	}
 
 	mapping := getDocMapping()
-	for _, tt := range tests {
+	for i := range tests {
+		tt := tests[i]
```

Comment:
question: Building upon my other comment, I did spend some time looking at this, and if you trim this test down to just the last usecase, and run it a bunch of times logging the output you can see that the output does vary.  I am pretty convinced that there is an issue with the production code, and I think it is flaky due to the randomised map iteration order.